homeassistant/components/sensor/repetier.py

```diff
@@ -0,0 +1,201 @@
+"""
+Sensor for retrieving Repetier-Server device status.
+
+Creates one sensor for each device attached to Repetier-Server
+Created by Morten Trab (morten@trab.dk) - 2019
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_PORT,
+    CONF_URL,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pyrepetier==1.4.1']
+
+__version__ = '0.5.1'
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_PORT, default='3344'): cv.string,
+    vol.Optional('decimals', default=1): int,
+    vol.Optional('state_percent', default=False): bool,
+})
+
+
+def parse_repetier_api_response(response):
+    """Parse the Repetier Server API json response."""
+    data_dict = {}
+
+    i = 0
+    for printer in response['data']:
+        _key = i
+        i += 1
+        if _key not in data_dict.keys():
+            data_dict[_key] = printer
+        else:
+            data_dict[_key].update(printer)
+    return data_dict
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Repetier sensors."""
+    import pyrepetier
+    try:
+        data = pyrepetier.RepetierData(
+            parse_repetier_api_response,
+            url=config.get(CONF_URL),
```

Comment:
Use the `[]` syntax for accessing required config options, `.get()` for the optional ones.